File: sample-infrastructure-services-domain/npm/src/commands/add-something.ts
import { inject, injectable } from "inversify";
import { ISampleUnitOfWork, ISampleUnitOfWorkFactory, Parent, TYPES as Domain } from "sample-domain";
import { AddSomethingCommand, IAddSomethingHandler } from "sample-services";
import { DomainHandlerBase } from "../domain-handler";

@injectable()
export class DomainAddSomethingHandler extends DomainHandlerBase<AddSomethingCommand, number> implements IAddSomethingHandler {

    readonly commandType = AddSomethingCommand.TYPE;

    constructor(@inject(Domain.Aggregates.ISampleUnitOfWorkFactory) uowFactory: ISampleUnitOfWorkFactory) {
        super(uowFactory);
    }

    protected async onHandleAsync(uow: ISampleUnitOfWork, command: AddSomethingCommand): Promise<number> {
        const newRecord = new Parent(command.someNewData);
        newRecord.addChild(newRecord.description);
        var inserted = await uow.parents.insertAsync(newRecord);
        return inserted.id;
    }

}

File: sample-infrastructure-services-domain/dotnet/src/commands/add-something.cs
using System.Threading.Tasks;
using Sample.Domain.Aggregates;
using Sample.Domain.Aggregates.Parent;
using Sample.Services.Commands;

namespace Sample.Infrastructure.Services.Domain.Commands
{
    public class DomainAddSomethingHandler : DomainHandlerBase<AddSomethingCommand, int>, IAddSomethingHandler
    {
        public DomainAddSomethingHandler(ISampleUnitOfWorkFactory uowFactory) : base(uowFactory) { }
        
        public override string CommandType => AddSomethingCommand.TYPE;

        protected override async Task<int> OnHandleAsync(ISampleUnitOfWork uow, AddSomethingCommand command) 
        {
            var newRecord = new Parent(command.SomeNewData);
            newRecord.AddChild(newRecord.Description);
            var inserted = await uow.Parents.InsertAsync(newRecord);
            return inserted.Id;
        }
    }
}
